File: dataset/avatar/Python/Code/codeforces_456_A.py
n = int ( input ( ) )
m = [ [ int ( x ) for x in input ( ).split ( ) ] for i in range ( n ) ]
m.sort ( reverse = True )
a = m [ 0 ]
for i in m [ 1 : ] :
    if i [ 1 ] > a [ 1 ] :
        print ( 'Happy Alex' )
        break
    a = i
else :
    print ( 'Poor Alex' )

File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_456_A.java

import java.util.Arrays;
import java.util.Comparator;
import java.util.Scanner;

public class codeforces_456_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());
        int[][] m = new int[n][2];
        for (int i = 0; i < n; i++) {
            String[] input = sc.nextLine().split(" ");
            m[i][0] = Integer.parseInt(input[0]);
            m[i][1] = Integer.parseInt(input[1]);
        }
        Arrays.sort(m, Comparator.comparingInt((int[] arr) -> arr[0]).reversed());
        int[] a = m[0];
        for (int i = 1; i < n; i++) {
            if (m[i][1] > a[1]) {
                System.out.println("Happy Alex");
                return;
            }
            a = m[i];
        }
        System.out.println("Poor Alex");
    }
}

